File: tp-arithmetic-java/src/test/java/fr/isima/controller/ArithmeticControllerTest.java
package fr.isima.controller;

import org.junit.Test;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;

public class ArithmeticControllerTest {

    private ArithmeticController arithmeticController = new ArithmeticController();

    @Test
    public void diviser_5par2_returns2dot5() {
        assertThat(this.arithmeticController.divide(5, 2)).isIn("2,5", "2.5");
    }

    @Test
    public void soustraire_5par2_returns3() {
        assertThat(this.arithmeticController.soustraire(5, 2)).isEqualTo("3");
    }

    @Test
    public void additionner_5par2_returns7() {
        assertThat(this.arithmeticController.additionner(5, 2)).isEqualTo("7");
    }

    @Test
    public void multiplier_5par2_returns10() {
        assertThat(this.arithmeticController.multiplier(5, 2)).isEqualTo("10");
    }
}

File: tp-arithmetic-kotlin/src/test/kotlin/fr/isima/controller/ArithmeticControllerTest.kt
package fr.isima.controller

import org.junit.Test

import org.assertj.core.api.AssertionsForClassTypes.assertThat

class ArithmeticControllerTest {

    private val arithmeticController = ArithmeticController()

    @Test
    fun diviser_5par2_returns2dot5() {
        assertThat(this.arithmeticController.divide(5.0, 2.0)).isIn("2,5", "2.5")
    }

    @Test
    fun soustraire_5par2_returns3() {
        assertThat(this.arithmeticController.soustraire(5.0, 2.0)).isEqualTo("3")
    }

    @Test
    fun additionner_5par2_returns7() {
        assertThat(this.arithmeticController.additionner(5.0, 2.0)).isEqualTo("7")
    }

    @Test
    fun multiplier_5par2_returns10() {
        assertThat(this.arithmeticController.multiplier(5.0, 2.0)).isEqualTo("10")
    }
}